src/main/scala/scorex/core/network/NetworkController.scala

```diff
@@ -37,6 +38,10 @@ class NetworkController(settings: NetworkSettings,
 
   private implicit val system: ActorSystem = context.system
 
+  private val featureSerializers: PeerFeature.Serializers = features.map(f => f.featureId -> f.serializer).toMap
+  private val handshakeSerializer = new HandshakeSerializer(featureSerializers, settings.maxHandshakeSize)
+
+  //todo: make usage more clear
```

Comment:
I think you'd better write more comments here, because it's not so clear, what should be improved (and this will be forgotten in a future)

Reply:
@catena2w I've addressed the issues reported, please make another pass!